### PR TITLE
Update Python dependencies

### PR DESCRIPTION
**Pull request by galaxybot**
Run `make update-dependencies`.

### CODE REVIEW

**Comment by nsoranzo** [comment]
Now that sphinx-rtd-theme 1.2.0 supports docutils 0.18, we are hitting https://github.com/galaxyproject/galaxy/issues/13005 again for the docs build.

**Comment by nsoranzo** [comment]
@davelopez @mvdbeek Can you please help with the 11 remaining new mypy errors reported in the Python linting build? They are all about the use of the `BaseModel.construct()` pydantic method. I've unsuccessfully tried to fix them for some hours, so better to leave these to the experts :)

**Comment by davelopez** [comment]
I'll take a look and try to fix them :+1:

**Comment by davelopez** [comment]
Hmm, making mypy happy seems to break everything else... I will take a closer look later today when I can.

**Comment by nsoranzo** [comment]
Depends on https://github.com/galaxyproject/galaxy/pull/15829 and https://github.com/galaxyproject/galaxy/pull/15844 .

@davelopez The galaxybot overwrote our commits with the updated dependency changes (it does that once a week), so I have rebased them.

**Comment by nsoranzo** [comment]
Test failures unrelated, this is ready for review/merge at last!

File: lib/galaxy/managers/folders.py [hunk 1]
```diff
@@ -539,17 +539,17 @@ def _filter_by_include_deleted(
...
+        self, sa_session: galaxy_scoped_session, folder: model.LibraryFolder
     ) -> List[Tuple[str, str]]:
         """
         Returns the folder path from root to the given folder.
 
         The path items are tuples with the name and id of each folder for breadcrumb building purposes.
         """
         current_folder = folder
-        path_to_root = [(f"F{security.encode_id(current_folder.id)}", current_folder.name)]
+        path_to_root = [(LibraryFolderDatabaseIdField.encode(current_folder.id), current_folder.name)]
```
Comment (jmchilton) [inline lib/galaxy/managers/folders.py:550]: -0.9 - depending on FastAPI at the top of the stack to create smart objects makes sense but forcing it deep in managers completely bypasses all our abstractions and renders code that is meant to be useable outside the web server context only useful when coming from FastAPI. Transformations like this are inappropriate IMO.

Comment (davelopez) [reply lib/galaxy/managers/folders.py:550]: This certainly can be reverted, but afaik `LibraryFolderDatabaseIdField` is not directly related to FastAPI, it is part of the schema module which managers already depend on. I thought the idea was to completely get rid of the manual encoding/decoding scattered across the entire code base and especially the managers and deal with it at the API layer so managers work always with simple integers.

But I agree this particular case was just a naive attempt to reduce manual handling of the `F` encoding for folders, but ideally, there will be no encoding at all here.

**Pull request merged**